src/collar/oc_anim.lsl

```diff
@@ -88,593 +89,782 @@ string g_sSettingToken = "anim_";
 //string g_sGlobalToken = "global_";
 key g_kWearer;
 
-list g_lMenuIDs;  //three strided list of avkey, dialogid, and menuname
+list g_lMenuIDs; // Three strided list of kAvatar, kDialogID, sMenuName
 integer g_iMenuStride = 3;
+
 /*
 integer g_iProfiled;
 Debug(string sStr) {
-    //if you delete the first // from the preceeding and following  lines,
-    //  profiling is off, debug is off, and the compiler will remind you to
-    //  remove the debug calls from the code, we're back to production mode
-    if (!g_iProfiled){
-        g_iProfiled=1;
-        llScriptProfiler(1);
-    }
-    llOwnerSay(llGetScriptName() + "(min free:"+(string)(llGetMemoryLimit()-llGetSPMaxMemory())+")["+(string)llGetFreeMemory()+"] :\n" + sStr);
+  // If you delete the first // from the preceeding and following  lines,
+  // profiling is off, debug is off, and the compiler will remind you to
+  // remove the debug calls from the code, we're back to production mode
+  if (!g_iProfiled) {
+    g_iProfiled = 1;
+    llScriptProfiler(1);
+  }
+  llOwnerSay(llGetScriptName() + "(min free:" + (string)(llGetMemoryLimit() - llGetSPMaxMemory()) + ")[" + (string)llGetFreeMemory() + "] :\n" + sStr);
 }
 */
 
 Dialog(key kID, string sPrompt, list lChoices, list lUtilityButtons, integer iPage, integer iAuth, string sName) {
-    key kMenuID = llGenerateKey();
-    llMessageLinked(LINK_DIALOG, DIALOG, (string)kID + "|" + sPrompt + "|" + (string)iPage + "|" + llDumpList2String(lChoices, "`") + "|" + llDumpList2String(lUtilityButtons, "`") + "|" + (string)iAuth, kMenuID);
+  key kMenuID = llGenerateKey();
+  llMessageLinked(LINK_DIALOG, DIALOG, (string)kID + "|" + sPrompt + "|" + (string)iPage + "|" + llDumpList2String(lChoices, "`") + "|" + llDumpList2String(lUtilityButtons, "`") + "|" + (string)iAuth, kMenuID);
+
+  integer iIndex = llListFindList(g_lMenuIDs, [kID]);
+  if (~iIndex) {
+    g_lMenuIDs = llListReplaceList(g_lMenuIDs, [kID, kMenuID, sName], iIndex, iIndex + g_iMenuStride - 1);
+  } else {
+    g_lMenuIDs += [kID, kMenuID, sName];  // We've not already given this user a menu. append to list
+  }
+  //Debug("Made " + sName + " menu.");
+}
 
-    integer iIndex = llListFindList(g_lMenuIDs, [kID]);
-    if (~iIndex) g_lMenuIDs = llListReplaceList(g_lMenuIDs, [kID, kMenuID, sName], iIndex, iIndex + g_iMenuStride - 1);
-    else g_lMenuIDs += [kID, kMenuID, sName];  //we've not already given this user a menu. append to list
-    //Debug("Made "+sName+" menu.");
+string Checkbox(integer iValue, string sLabel) {
+  if (iValue) return "☑ " + sLabel;
```

Comment:
I see in other places you've added curly braces to `if` statements that previously lacked them.  Could you do that here too?

Reply:
Yes, but... I think it can be omitted for simple one-liners, like those you shown.
Just compare short variant:
```lsl
else if (sToken == "PoseMoveRun") g_sPoseMoveRun = sValue;
else if (sToken == "PostureRank") g_iLastPostureRank = (integer)sValue;
else if (sToken == "PoselockRank") g_iLastPoselockRank = (integer)sValue;
```
and full:
```lsl
else if (sToken == "PoseMoveRun") {
  g_sPoseMoveRun = sValue;
} else if (sToken == "PostureRank") {
  g_iLastPostureRank = (integer)sValue;
} else if (sToken == "PoselockRank") {
  g_iLastPoselockRank = (integer)sValue;
}
```
Doesn't short variant looks better for such places?

Reply:
The curly braces on one liners looks messier than the one liner does, it's not too difficult to add curly braces if you need to add another action to that if statement. :)